src/Util/BatchRequest.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Http\Client\Util;
+
+use Http\Client\BatchResult;
+use Http\Client\Exception;
+use Http\Client\Exception\BatchException;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Implements sending multiple request for client not supporting parallels requests
+ *
+ * Internally, use the sendRequest method
+ *
+ * Should be used with Http\Client\HttpPsrClient
+ *
+ * @author Joel Wurtz <jwurtz@jolicode.com>
+ */
+trait BatchRequest
+{
+    /**
+     * {@inheritdoc}
+     */
+    abstract public function sendRequest(RequestInterface $request, array $options = []);
```

Comment:
This problem can be handled on a test level: if the stub implements HttpPsrClient then the tests will fail if the signature is incorrect.